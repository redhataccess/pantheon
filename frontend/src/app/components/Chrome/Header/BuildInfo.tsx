import React, { Component } from 'react';
import { Link } from "react-router-dom";

class BuildInfo extends Component {
    public state = {
        buildDate: '',
        commitHash: '',
        commitText: ''
    };

    public componentDidMount() {
        this.getBuildInfo()
    }

    public render() {
        return (
            <React.Fragment>
                <div className="column-view">
                    {!this.state.commitHash.includes("OPENSHIFT_BUILD_COMMIT") && <Link to=""
                        onClick={this.commitRedirect}>
                        {this.state.buildDate}
                    </Link>}
                    {this.state.commitHash.includes("OPENSHIFT_BUILD_COMMIT") && this.state.buildDate}
                </div>
            </React.Fragment>
        );
    }

    public getBuildInfo() {
        const backend = "/pantheon/builddate.json"
        if (this.state.buildDate === '') {
            fetch(backend)
                .then(response => response.json())
                .then(responseJSON => this.setState({ buildDate: "Build Date: " + responseJSON.buildDate, commitHash: responseJSON.commitHash }))
        }
    };

    private commitRedirect = () => {
        window.location.href = "https://github.com/redhataccess/pantheon/commit/" + this.state.commitHash
    }
}

export { BuildInfo }