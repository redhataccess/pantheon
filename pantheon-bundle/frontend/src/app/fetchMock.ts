if (!self.fetch) {
    self.fetch = (input: RequestInfo, init?: RequestInit | undefined) => {
        const p = new Promise<Response>((resolve, reject) => {
            // just don't do anything. this exists so that our tests don't try and actually resolve fetches,
            // (which is exactly what isomorphic-fetch tried to do while we were importing it...)
            // and calling nothing here allows us to skip all of our fetch.then() and fetch.catch() code.
        })
        return p
    }
}