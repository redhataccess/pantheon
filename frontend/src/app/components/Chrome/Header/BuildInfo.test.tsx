import React from 'react'
import { BuildInfo }  from './BuildInfo'
import '@app/fetchMock'

import { shallow } from 'enzyme'

describe('BuildInfo tests', () => {
  test('should render NavLinks component', () => {
    const view = shallow(<BuildInfo />)
    expect(view).toMatchSnapshot()
  })

  test('getBuildInfo call', () => {
    const spy = jest.spyOn(BuildInfo.prototype, 'getBuildInfo')
    shallow(<BuildInfo />)

    expect(spy).toHaveBeenCalled()
  })

})