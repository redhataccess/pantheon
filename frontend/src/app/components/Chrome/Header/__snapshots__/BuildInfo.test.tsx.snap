// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildInfo tests should render NavLinks component 1`] = `
<Fragment>
  <div
    className="column-view"
  >
    <Link
      onClick={[Function]}
      to=""
    />
  </div>
</Fragment>
`;
